.github/workflows/ci.yaml

```diff
@@ -0,0 +1,53 @@
+name: Rust CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  build_and_test:
+    name: Build and test
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            rust: stable
+
+          # Minimum Supported Rust Version (MSRV)
+          - os: ubuntu-latest
+            rust: 1.60.0
+
+    runs-on: ${{ matrix.os }}
+    steps:
+    - name: Install dependencies
+      run: sudo apt install alsa libasound2-dev libxcb-shape0-dev libxcb-xfixes0-dev
+
+    - name: Install Rust toolchain
+      uses: dtolnay/rust-toolchain@master
+      with:
+        # We have to use `master` and pass the toolchain explicitly
+        toolchain: ${{ matrix.rust }}
+
+    - name: Check out repository
+      uses: actions/checkout@v3
+
+    - name: Generate Cargo.lock
+      run: cargo generate-lockfile
```

Comment:
I don't think running this on CI practically does anything. If the latest versions compatible with specifications in Cargo.toml are not compatible with the MSRV, the build will fail:

```
creek on  ci_workspace [$✘?] is 📦 v1.0.0 via 🦀 v1.72.0-nightly 
❯ cargo +1.56-x86_64-unknown-linux-gnu generate-lockfile
    Updating crates.io index

creek on  ci_workspace [$!?] is 📦 v1.0.0 via 🦀 v1.72.0-nightly 
❯ cargo +1.56-x86_64-unknown-linux-gnu check
error: package `log v0.4.19` cannot be built because it requires rustc 1.60.0 or newer, while the currently active rustc version is 1.56.1
```

Reply:
Not committing Cargo.lock and generating during the CI build is common practice for library crates.

Reply:
The Cargo.lock file is only needed for caching.

Versioning the Cargo.lock file just for the MSRV CI build is not suitable.